sky/clouds/cloud_registry.py

```diff
@@ -1,31 +1,81 @@
 """Clouds need to be registered in CLOUD_REGISTRY to be discovered"""
 
 import typing
-from typing import Optional, Type
+from typing import Callable, Dict, List, Optional, overload, Type, Union
 
 from sky.utils import ux_utils
 
 if typing.TYPE_CHECKING:
     from sky.clouds import cloud
 
 
-class _CloudRegistry(dict):
+class _CloudRegistry:
     """Registry of clouds."""
 
+    def __init__(self) -> None:
+        self.clouds: Dict[str, 'cloud.Cloud'] = {}
+        self.aliases: Dict[str, str] = {}
+
     def from_str(self, name: Optional[str]) -> Optional['cloud.Cloud']:
         if name is None:
             return None
-        if name.lower() not in self:
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError(f'Cloud {name!r} is not a valid cloud among '
-                                 f'{list(self.keys())}')
-        return self.get(name.lower())
 
+        search_name = name.lower()
+
+        if search_name in self.clouds:
+            return self.clouds[search_name]
+
+        if search_name in self.aliases:
+            return self.clouds[self.aliases[search_name]]
+
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(f'Cloud {name!r} is not a valid cloud among '
+                             f'{[*self.clouds.keys(), *self.aliases.keys()]}')
+
+    @staticmethod
+    def to_canonical_name(cloud: Optional['cloud.Cloud']) -> Optional[str]:
```

Comment:
Thanks for adding cloud.canonical_name()! With that, looks like this method is redundant and can be removed since callers can directly call cloud.canonical_name()?

Reply:
This one supports optional values, but that's the only difference.